bika/lims/browser/reports/productivity_analysesperclient.py

```diff
@@ -20,170 +26,167 @@ class Report(BrowserView):
+            client = analysis.getClientTitle
+            data_client = data.get(client, {})
+            request = analysis.getRequestID
+            requests = data_client.get("requests", [])
+            if request not in requests:
+                requests += [request]
+                data_client["requests"] = requests
+                total_num_ars += 1
+            data_client["analyses"] = data_client.get("analyses", 0) + 1
+            data[client] = data_client
+
+        # Generate datalines
+        data_lines = list()
+        for client, data in data.items():
+            ars_count = len(data.get('requests', []))
+            ans_count = data.get('analyses', 0)
+            data_lines.append([{"value": client},
+                               {"value": ars_count},
+                               {"value": ans_count}])
 
-        self.report_content = {}
-        parm_lines = {}
-        parms = []
-        headings = {}
-        count_all_ars = 0
-        count_all_analyses = 0
-        query = {}
-
-        this_client = logged_in_client(self.context)
-
-        if not this_client and 'ClientUID' in self.request.form:
-            client_uid = self.request.form['ClientUID']
-            this_client = rc.lookupObject(client_uid)
-            parms.append(
-                {'title': _('Client'),
-                 'value': this_client.Title(),
-                 'type': 'text'})
-
-        if this_client:
-            headings['header'] = _("Analysis requests and analyses")
-            headings['subheader'] = _("Number of Analysis requests and analyses")
-        else:
-            headings['header'] = _("Analysis requests and analyses per client")
-            headings['subheader'] = _(
-                "Number of Analysis requests and analyses per client")
+        if self.request.get('output_format', '') == 'CSV':
+            return self.generate_csv(data_lines)
 
+        self.report_content = {
+            'headings': self.headings,
+            'parms': parms,
+            'formats': self.formats,
+            'datalines': data_lines,
+            'footings': [
+                [{'value': _('Total'), 'class': 'total_label'},
+                 {'value': total_num_ars},
+                 {'value': total_num_analyses},],]
+        }
+
+        return {'report_title': t(self.headings['header']),
+                'report_data': self.template()}
+
+    def add_filter_by_client(self, query, out_params):
+        """Applies the filter by client to the search query
+        """
+        current_client = logged_in_client(self.context)
+        if current_client:
+            query['getClientUID'] = api.get_uid(current_client)
+        elif self.request.form.get("ClientUID", ""):
+            query['getClientUID'] = self.request.form['ClientUID']
+            client = api.get_object_by_uid(query['getClientUID'])
+            out_params.append({'title': _('Client'),
+                               'value': client.Title,
```

Comment:
`client` is an object here, so you need to call `Title()`

<img width="329" alt="27 09 2018 - analysis requests and analyses per client 1 seite 2018-09-27 16-45-25" src="https://user-images.githubusercontent.com/713193/46154103-bd234400-c274-11e8-95aa-58a6e4c88b1d.png">

Reply:
Ooooopsss..... thanks!